server.go

```diff
@@ -10,7 +10,8 @@ import (
 func main() {
 	s := server.NewServer("localhost")
 
-	s.Run()
+	config := server.Config{UseSmartAuth: true, UseLoggingMiddleware: true}
```

Comment:
I see now that even if SmartAuth is enabled, we let requests go through if there is no `X-DELEGATED` header -- so I guess I'm less concerned about having this enabled by default.

That said, I'm a bit surprised at this behavior (more on that later).